test/framework/executor_itf/executor_itf.cpp

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "framework/executor_itf/executor_itf.hpp"
+
+#include "ametsuchi/impl/postgres_command_executor.hpp"
+#include "ametsuchi/impl/postgres_query_executor.hpp"
+#include "ametsuchi/tx_executor.hpp"
+#include "backend/protobuf/proto_permission_to_string.hpp"
+#include "framework/config_helper.hpp"
+#include "framework/test_logger.hpp"
+#include "interfaces/permissions.hpp"
+#include "logger/logger.hpp"
+#include "logger/logger_manager.hpp"
+#include "module/shared_model/mock_objects_factories/mock_command_factory.hpp"
+#include "module/shared_model/mock_objects_factories/mock_query_factory.hpp"
+
+using namespace iroha::integration_framework;
+using namespace iroha::ametsuchi;
+using namespace common_constants;
+
+namespace {
+  logger::LoggerManagerTreePtr getDefaultLogManager() {
+    return getTestLoggerManager()->getChild("ExecutorITF");
+  }
+
+  template <typename OtherResult>
+  ExecutorItf::RunResult makeRunResult(const OtherResult &other_result) {
+    return other_result.match(
+        [](const auto &) -> ExecutorItf::RunResult { return {}; },
+        [](const auto &error) -> ExecutorItf::RunResult {
+          return error.error.toString();
+        });
+  }
+
+  std::string getDefaultRole(const std::string &name) {
+    return name + "_default_role";
+  }
+}  // namespace
+
+ExecutorItf::ExecutorItf()
+    : log_manager_(getDefaultLogManager()),
+      log_(log_manager_->getLogger()),
+      pg_connection_string_(
+          ::integration_framework::getPostgresCredsOrDefault()),
+      is_initialized_(false),
+      mock_command_factory_(
+          std::make_unique<shared_model::interface::MockCommandFactory>()),
+      mock_query_factory_(
+          std::make_unique<shared_model::interface::MockQueryFactory>()),
+      query_counter_(0) {}
+
+ExecutorItf::~ExecutorItf() {
+  //connection_manager_->dropDatabase();
```

Comment:
The commented code should either be removed or uncommented.

Reply:
> The commented code should either be removed or uncommented.

As I want to merge this big piece of code without using trunk, I need to omit some parts that are logically required. I created these comments to show what and where the missing components will be. This is intended to alleviate the review (you can see what the corresponding code is designed to do without actual implementation) and show that this is a work in progress (not ready to be used).

Reply:
Also, we have such comments in our project that are there for months, and these are going to be replaced with working code in the next PR. So I see no harm from them.

Reply:
>we have such comments in our project that are there for months

In this case they also have to be removed, does not looks like a valid reason for me. If the idea is to show the missing components, it might be better to add textual descriptions instead, like "database should be dropped here".